src/shared/components/comment/comment-node.tsx

```diff
@@ -137,7 +137,10 @@ export class CommentNode extends Component<CommentNodeProps, CommentNodeState> {
   state: CommentNodeState = {
     showReply: false,
     showEdit: false,
-    collapsed: false,
+    // Collapse comments that have no children and are removed by default
+    collapsed:
+      this.commentView.comment.child_count === 0 &&
```

Comment:
No need to change: since 0 is falsy, the left half of the and can be `!this.commentView.comment.child_count`.